api/app/src/command/medical/patient/create-patient.ts

```diff
@@ -1,19 +1,31 @@
-import { Patient } from "../../../models/medical/patient";
+import {
+  Patient,
+  PatientCreate as PatientCreateModel,
+  PatientData,
+} from "../../../models/medical/patient";
 
-export const createPatient = async ({
-  facilityId,
-  cxId,
-  data,
-}: {
-  facilityId: string;
-  cxId: string;
-  data: object;
-}): Promise<Patient> => {
-  return Patient.create({
+type Identifier = Pick<Patient, "cxId"> & { facilityId: string };
+type PatientNoExternalData = Omit<PatientData, "externalData">;
+type PatientCreate = PatientNoExternalData & Identifier;
+
+export const createPatient = async (patient: PatientCreate): Promise<Patient> => {
+  const { cxId, facilityId } = patient;
+
+  const newPatient: PatientCreateModel & Pick<Patient, "id"> = {
     id: "", // the patient id will be generated on the beforeCreate hook
+    patientNumber: 0, // this will be generated on the beforeCreate hook
     cxId,
     facilityIds: [facilityId],
-    patientNumber: 0, // this will be generated on the beforeCreate hook
-    data,
-  });
+    data: {
+      firstName: patient.firstName,
+      lastName: patient.lastName,
+      dob: patient.dob,
+      gender: patient.gender,
+      driversLicense: patient.driversLicense,
```

Comment:
Would this be better suited as an array of identities where it can be a License, SSN, or etc?